app/Controllers/Http/OrdersController.ts

```diff
@@ -0,0 +1,114 @@
+import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext';
+import Database from '@ioc:Adonis/Lucid/Database';
+import Order from 'App/Models/Order';
+import CreateOrder from 'App/Validators/CreateOrderValidator';
+import UpdateOrder from 'App/Validators/UpdateOrderValidator';
+
+export default class OrdersController {
+    public async index({}: HttpContextContract) {
+        const orders = await Database.from('orders')
+            .join('products', 'orders.product_id', '=', 'products.id')
+            .select('orders.id')
+            .select('orders.quantity')
+            .select('products.name')
+            .select('products.price')
+
+        if (Object.keys(orders).length === 0) {
+            return {
+                status: 'fail',
+                message: 'Failed to retrieve orders'
+            }
+        }
+
+        return {
+            status: 'success',
+            data: orders,
+            message: 'Orders retrieved successfully'
+        }
+    }
+
+    public async store({ request }) {
+        const order = new Order();
+        const payload = await request.validate(CreateOrder);
+
+        order.quantity = payload.quantity;
+        order.productId = payload.productId;
+
+        await order.save();
+
+        if (!order.$isPersisted) {
+            return {
+                status: 'fail',
+                message: 'Failed to create new order'
+            }
+        }
+
+        return {
+            status: 'success',
+            data: order,
+            message: 'Order created successfully'
+        }
+    }
+
+    public async show({ params }: HttpContextContract) {
+        const order = await Database.from('orders')
+            .join('products', 'orders.product_id', '=', 'products.id')
+            .select('orders.id')
+            .select('orders.quantity')
+            .select('products.name')
+            .select('products.price')
+            .where('orders.id', params.id)
+
+        if (Object.keys(order).length === 0) {
+            return {
+                status: 'fail',
+                message: 'Failed to retrieve order whose id is ' + params.id
+            }
+        }
+
+        return {
+            status: 'success',
+            data: order,
+            message: 'Order retrieved successfully'
+        }
+    }
+
+    public async update({ params, request }: HttpContextContract) {
+        const order = await Order.findOrFail(params.id); 
```

Comment:
There is no try catch here? Is all of this code error safe?

Reply:
I will clean up using the try-catch here, however, there is this piece of code **findOrFail(params.id);**, the **find()** method finds a record using the primary key. That method returns a model instance or null when no records are found. So with **findOrFail()**, the orFail variation, raises an exception when no row in the database table is found.

Reply:
Ok ok, then it is not problem :D 
Will it survive a connection failure aswell? Or will that be brought into a try catch?

Reply:
Yes, it will survive a connection failure.